### ISSUE TITLE
real order countdown

### ISSUE THREAD

**Issue by wille** [bitrefill/airfill-widget#129]

**Comment by ziggamon** [comment]
There is code for this in the old bitrefill-web.

**Comment by wille** [comment]
@ziggamon code that calculates expiry time for order is run on backend in bitrefill-web. if user has misconfigured timezone it will show invalid values

**Comment by ziggamon** [comment]
Yes. This is a problem we have already solved. Timezones are not an issue because we don't care about user's clock at all. 

In any case this isn't a huge problem as the clock is mainly a UX element not tied to a specific functionality, but of course it would be good if it did the right thing :)

**Comment by ziggamon** [comment]
See bitrefill-web, routes/views/order.js L92: 

```
function expiryTimeForOrder(order) {
	var timeLeft = order.expirationTime - Date.now();
	// Round to nearest second (or 0)
	return Math.ceil(Math.max(timeLeft / 1000, 0));
}
```

**Comment by wille** [comment]
yes that code is ran on the server serving bitrefill-web, in client side Date.now()  will fuck up when comparing to order.expirationTime if the user has misconfigured time

**Comment by ziggamon** [comment]
Yes, this is why you should run the code serverside. I'm pretty sure there's been discussion of this already, if it hasn't even already been implemented in -backend.

**Comment by ziggamon** [comment]
> if it hasn't even already been implemented in -backend

Seems not, but easily done

**Comment by wille** [comment]
Great.
